src/components/ChangePlan/Checkout/BillingInformation/BillingInformation.js

```diff
@@ -0,0 +1,297 @@
+import React, { useEffect, useState } from 'react';
+import { InjectAppServices } from '../../../../services/pure-di';
+import { useIntl } from 'react-intl';
+import useTimeout from '../../../../hooks/useTimeout';
+import {
+  FieldGroup,
+  FieldItem,
+  InputFieldItem,
+  PhoneFieldItem,
+  SelectFieldItem,
+  CheckboxFieldItem,
+  SubmitButton,
+} from '../../../form-helpers/form-helpers';
+import { Form, Formik } from 'formik';
+import { getCountries, getFormInitialValues } from '../../../../utils';
+import { Loading } from '../../../Loading/Loading';
+
+const fieldNames = {
+  sameAddressAsContact: 'sameAddressAsContact',
+  firstname: 'firstname',
+  lastname: 'lastname',
+  address: 'address',
+  city: 'city',
+  province: 'province',
+  country: 'country',
+  zipCode: 'zipCode',
+  phone: 'phone',
+};
+
+export const BillingInformation = InjectAppServices(
+  ({
+    dependencies: { dopplerUserApiClient, dopplerBillingUserApiClient, staticDataClient },
+    handleSaveAndContinue,
+    showTitle,
+  }) => {
+    const [state, setState] = useState({ loading: true });
+    const [formSubmitted, setFormSubmitted] = useState(false);
+    const [states, setStates] = useState([]);
+    const [sameAddressInformation, setSameAddressInformation] = useState(false);
+    const createTimeout = useTimeout();
+    const intl = useIntl();
+
+    useEffect(() => {
+      const fetchData = async () => {
+        const contactInformationResult = await dopplerUserApiClient.getContactInformationData();
+        const billingInformationResult =
+          await dopplerBillingUserApiClient.getBillingInformationData();
+        const statesResult = await staticDataClient.getStatesData(
+          billingInformationResult.success ? billingInformationResult.value.country : 'ar',
+          intl.locale,
+        );
+
+        setStates(statesResult.success ? statesResult.value : []);
+
+        const billingInformation = billingInformationResult.success
+          ? billingInformationResult.value
+          : { sameAddressAsContact: true };
+        if (billingInformation && billingInformation.sameAddressAsContact) {
```

Comment:
```suggestion
if (billingInformation?.sameAddressAsContact) {
```

Reply:
A function could be used that, given a `contactInformationResult.value`, returns the billing data. To make the code a little more declarative. Something like this: 
```
billingInformation = {
    ...getBillingInformation(contactInformationResult.value),
}
```

Reply:
👍